File: StackExchange Client/src/java/controllers/VoteController.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package controllers;

import AnswerWS.AnswerWS_Service;
import QuestionWS.QuestionWS_Service;
import UserWS.User;
import connector.ISConnector;
import java.io.IOException;
import javax.servlet.ServletException;
import javax.servlet.http.Cookie;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import javax.xml.ws.WebServiceRef;
import org.json.simple.JSONObject;

/**
 *
 * @author vanyadeasysafrina
 */
public class VoteController extends HttpServlet {
    @WebServiceRef(wsdlLocation = "WEB-INF/wsdl/localhost_8081/StackExchange_WebService/AnswerWS.wsdl")
    private AnswerWS_Service service_1;
    @WebServiceRef(wsdlLocation = "WEB-INF/wsdl/localhost_8081/StackExchange_WebService/QuestionWS.wsdl")
    private QuestionWS_Service service;

    /**
     * Processes requests for both HTTP <code>GET</code> and <code>POST</code>
     * methods.
     *
     * @param request servlet request
     * @param response servlet response
     * @throws ServletException if a servlet-specific error occurs
     * @throws IOException if an I/O error occurs
     */
    protected void processRequest(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        User user = (User) request.getAttribute("user");
        
        String type = request.getParameter("type");
        String id = request.getParameter("id");
        String up = request.getParameter("up");
        
        if(user!=null) {
            HttpServletRequest httpRequest = (HttpServletRequest) request;
            Cookie[] cookies = httpRequest.getCookies();
            String status = "No cookie";
            // Check cookie with name auth
            if (cookies != null) {
                String token = null;
                for (Cookie cookie : cookies) {
                    status = "No token cookie";
                    if (cookie.getName().equals("token")) {
                        token = cookie.getValue();
                        break;
                    }
                }
            }
            //TODO:
            JSONObject result = ISConnector.vote(type, id, type, up);
            response.sendRedirect("question?q_id="+id);
        }
        else {
            String message = "Please log in before ";
            String url = request.getContextPath() + "question?q_id=" + id;
            request.setAttribute("message", message);
            request.setAttribute("url", url);
            response.sendRedirect(request.getContextPath() + "/login?st=0");
        }
    }

    // <editor-fold defaultstate="collapsed" desc="HttpServlet methods. Click on the + sign on the left to edit the code.">
    /**
     * Handles the HTTP <code>GET</code> method.
     *
     * @param request servlet request
     * @param response servlet response
     * @throws ServletException if a servlet-specific error occurs
     * @throws IOException if an I/O error occurs
     */
    @Override
    protected void doGet(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        processRequest(request, response);
    }

    /**
     * Handles the HTTP <code>POST</code> method.
     *
     * @param request servlet request
     * @param response servlet response
     * @throws ServletException if a servlet-specific error occurs
     * @throws IOException if an I/O error occurs
     */
    @Override
    protected void doPost(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        processRequest(request, response);
    }

    /**
     * Returns a short description of the servlet.
     *
     * @return a String containing servlet description
     */
    @Override
    public String getServletInfo() {
        return "Short description";
    }// </editor-fold>

    private String voteQuestion(int qId, int uId) {
        // Note that the injected javax.xml.ws.Service reference as well as port objects are not thread safe.
        // If the calling of port operations may lead to race condition some synchronization is required.
        QuestionWS.QuestionWS port = service.getQuestionWSPort();
        return port.voteQuestion(qId, uId);
    }

    private String devoteQuestion(int qId, int uId) {
        // Note that the injected javax.xml.ws.Service reference as well as port objects are not thread safe.
        // If the calling of port operations may lead to race condition some synchronization is required.
        QuestionWS.QuestionWS port = service.getQuestionWSPort();
        return port.devoteQuestion(qId, uId);
    }

    private String voteAnswer(int aId, int uId) {
        // Note that the injected javax.xml.ws.Service reference as well as port objects are not thread safe.
        // If the calling of port operations may lead to race condition some synchronization is required.
        AnswerWS.AnswerWS port = service_1.getAnswerWSPort();
        return port.voteAnswer(aId, uId);
    }

    private String devoteAnswer(int aId, int uId) {
        // Note that the injected javax.xml.ws.Service reference as well as port objects are not thread safe.
        // If the calling of port operations may lead to race condition some synchronization is required.
        AnswerWS.AnswerWS port = service_1.getAnswerWSPort();
        return port.devoteAnswer(aId, uId);
    }

}

File: StackExchange_Client/web/js/controllers/voteController.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
app.controller('questionVoteController', [ '$scope','$http','$location','$cookies', function($scope,$http,$location,$cookies)
    {
        var questionvotecontrol = this;
        $scope.vote = 0;
        var votedata = this.votedata;
        this.votedata = {};
        var qid = $location.search()['id'];
            $http(
                {
                    url: "http://localhost:8080/StackExchange_Client/vote",
                    method: "GET",
                    params: {id: qid, db: "question"}
                }).success(function(data)
                {
                    if (data.vote !== null)
                    {
                        $scope.vote = data.vote;
                    }
                }    
                    );
        $scope.addVote = function(x){
            var token = $cookies.get("stackexchange_token");
            if (token!==null) {
                $http({
                    url: "http://localhost:8080/StackExchange_Client/addVote",
                    method: "POST",
                    data: {
                        id: qid,
                        vote: x,
                        db: "question",
                        token: token
                    },
                     headers: {'Content-Type': 'application/x-www-form-urlencoded'},
                  transformRequest: function(obj) {
                    var str = [];
                    for(var p in obj)
                    str.push(encodeURIComponent(p) + "=" + encodeURIComponent(obj[p]));
                    return str.join("&");
                  },
                }).success(function(data){
                    if (data.ERROR==null) {
                        $scope.vote = data.vote;
                    }
                    else {
                        alert(data.ERROR);
                    }
                });
            }
                else {
                    alert("You must log in to vote.");
                }
        };
        this.init = function(qid)
        {
            alert("tes");
            this.votedata.qid = qid;
        };
        
    }
]);